src/sentry/stacktraces/functions.py

```diff
@@ -271,6 +272,29 @@ def get_function_name_for_frame(frame, platform=None):
         return trim_function_name(rv, frame.get("platform") or platform)
 
 
+def get_source_link_for_frame(frame: dict[str, Any]) -> str:
+    """If source_link points to a GitHub raw content link, process it so that
+    we can return the GitHub equivalent with the line number, and use it as a
+    stacktrace link. Otherwise, return the link as is.
+    """
+    source_link = frame.get("source_link")
+    parse_result = urlparse(source_link)
+    if parse_result.netloc == "raw.githubusercontent.com":
+        path_parts = parse_result.path.split("/")
+        if path_parts[0] == "":
+            # the path starts with a "/" so the first element is empty string
+            del path_parts[0]
+
+        # at minimum, the path must have an author/org, a repo, and a file TODO: how to detect the last element is a valid file (don't want to link to a folder)
```

Comment:
from what i could tell, there are no raw github user content folders or repos (only files) so maybe we don't need to worry about detecting that the last element is a valid file?